docs/content/src/learn/tutorials/messaging-integrations/messaging-with-kafka/README.md

```diff
@@ -0,0 +1,272 @@
+
+Let us now see the complete implementation of the `product_admin_service`, which is a Kafka topic publisher. This publisher accepts HTTP requests and routes the serialized message to a partition in the topic based on the content of the recieved payload. In this case it filters for product type.
+
+#### product_admin_service.bal
+
+```ballerina
+import ballerina/http;
+import ballerina/kafka;
+import ballerina/io;
+
+// Kafka Producer Configuration
+kafka:ProducerConfig producerConfigs = {
+    bootstrapServers: "localhost:9092",
+    clientId: "kafka-producer",
+    acks: "all",
+    retryCount: 3
+};
+
+// Kafka Producer
+kafka:Producer kafkaProducer = new(producerConfigs);
+
+// HTTP Service Endpoint
+listener http:Listener httpListener = new(9090);
+
+@http:ServiceConfig { basePath: "/product" }
+service productAdminService on httpListener {
+
+    @http:ResourceConfig { methods: ["POST"], consumes: ["application/json"], produces: ["application/json"] }
+    resource function updatePrice(http:Caller caller, http:Request request) returns error? {
+        http:Response response = new;
+
+        json reqPayload = check request.getJsonPayload();
+        io:println("ProductManagementService : Received payload");
+
+        var productName = reqPayload.Product;
+        var productType = reqPayload.Type;
+        var productPrice = reqPayload.Price;
+
+        // Construct message to be published to the Kafka Topic
+        json productInfo = {
+            "Name" : productName.toString(),
+            "Price" : productPrice.toString()
+        };
+
+        // Serialize the message
+        byte[] serializedMessage = productInfo.toJsonString().toBytes();
+
+        if (productType.toString() == "Fruit") {
+            io:println("ProductManagementService : Sending message to Partition 0");
+            var sendResult = kafkaProducer->send(serializedMessage, "product-price", partition = 0);
+        } else if (productType.toString() == "Vegetable") {
+            io:println("ProductManagementService : Sending message to Partition 1");
+            var sendResult = kafkaProducer->send(serializedMessage, "product-price", partition = 1);
+        }
+
+        response.setJsonPayload({ "Status" : "Success" });
+        var responseResult = caller->respond(response);
+    }
+}
+```
+
+Let us consider the consumer services, `consumer1` and `consumer2`
+
+```ballerina
+// Kafka Consumer Configuration
+kafka:ConsumerConfig consumer1 = {
+    bootstrapServers: "localhost:9092",
+    groupId: "consumer",
+    topics: ["product-price"],
+    pollingIntervalInMillis: 1000,
+    partitionAssignmentStrategy: "org.apache.kafka.clients.consumer.RoundRobinAssignor"
+
+    listener kafka:Consumer productConsumer1 = new (consumer1);
+};
+```
+
+We make use of a partitionAssignmentStrategy to assign the consumer to a particular topic partition.
```

Comment:
```suggestion
We make use of a `partitionAssignmentStrategy` to assign the consumer to a particular topic partition.
```
Use single backticks for such references.

Reply:
Done